src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.cs

```diff
@@ -420,7 +421,10 @@ private IEnumerable<string> CollectInputs(BuildUpToDateCheckLogger logger)
             if (_customInputs.Count != 0)
             {
                 logger.Verbose("Adding " + UpToDateCheckInput.SchemaName + " inputs:");
+                // TODO remove pragmas when https://github.com/dotnet/roslyn/issues/37040 is fixed
+#pragma warning disable CS8622
                 foreach (string input in _customInputs.Select(_configuredProject.UnconfiguredProject.MakeRooted))
+#pragma warning restore CS8622
```

Comment:
Filed https://github.com/dotnet/roslyn/issues/37040 for a few instances of this problem.

Reply:
That's a good catch, I would have assumed that any warning there would be because CPS is un-annotated.

Reply:
The NRT feature is designed to work when consuming un-annotated code. In general if you're seeing warnings that you don't agree with, it's worth digging in (at least while the feature is young).